rust/rope/src/rope.rs

```diff
@@ -708,6 +708,16 @@ impl fmt::Debug for Rope {
     }
 }
 
+impl fmt::Display for Rope {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if f.alternate() {
+            write!(f, "{}", String::from(self))
```

Comment:
Although this works, and is probably reasonable for most uses, I will observe that it isn't _ideal_; specifically it is unnecessarily allocating a new string each time. A more efficient approach would be to iterate through the nodes of the rope with `self.iter_chunks_all()`, and then `write!()` each chunk directly.